src/rules/property-shorthand-no-unnecessary-value/index.js

```diff
@@ -0,0 +1,104 @@
+import valueParser from "postcss-value-parser"
+import { vendor } from "postcss"
+import {
+  cssWordIsVariable,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "property-shorthand-no-unnecessary-value"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (unexpected, expected) => `Unexpected longhand value '${unexpected}' instead of '${expected}'`,
+})
+
+const shorthandableProperties = new Set([
+  "margin",
+  "padding",
+  "border-color",
+  "border-radius",
+  "border-style",
+  "border-width",
+])
+
+function isIgnoredCharacters(value) {
+  return value.indexOf("+") !== -1
```

Comment:
Oops, the `_.includes()` suggestion was indeed wrong. How about `some()`?

`return ignoredCharacters.some(char => value.indexOf(char) !== -1)`